libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/LocationEx.kt

```diff
@@ -48,11 +51,48 @@ internal fun Location.toFixLocation(): FixLocation {
         provider,
         bearingAccuracy,
         speedAccuracy,
-        verticalAccuracy
+        verticalAccuracy,
+        if (extras != null) extras.toMap() else Bundle().toMap()
     )
 }
 
 internal fun List<FixLocation>.toLocations(): List<Location> = this.map { it.toLocation() }
 
 private fun isCurrentSdkVersionEqualOrGreaterThan(sdkCode: Int): Boolean =
     Build.VERSION.SDK_INT >= sdkCode
+
+internal fun HashMap<String, Value>.toBundle(): Bundle? {
+    if (this.isEmpty()) return null
```

Comment:
This looks incorrect to me. 
When we convert empty Bundle to the HashMap we get empty HashMap, but when we convert same empty HashMap back to Bundle we get null.